granulate_utils/containers/container.py

```diff
@@ -0,0 +1,24 @@
+#
+# Copyright (c) Granulate. All rights reserved.
+# Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
+#
+
+from dataclasses import dataclass
+from typing import Dict, Optional
+
+
+@dataclass
+class Container:
+    """
+    Shared "Container" descriptor class, used for Docker containers & CRI containers.
+    """
+
+    runtime: str  # docker / containerd / crio
+    # container name for Docker
+    # reconstructed container name (as if it were Docker) for CRI
+    name: str
+    id: str
+    labels: Dict[str, str]
+    running: bool
+    # None if not requested / container is dead
+    pid: Optional[int]
```

Comment:
What about `namespace: str `, `env_vars: list` we don't want to keep that too ?

Reply:
`env` - currently unused
`namespace` - can be derived from the `name` and exists in `labels`, no need to maintain a separate field.